cookbook/sortAbsoluteInstants.mjs

```diff
@@ -0,0 +1,33 @@
+import assert from "assert";
+
+/**
+ * getSortedInstants will sort an array of strings (each of which is parseable as a Temporal.Absolute and may or may not include an IANA time zone name)
+ * by the corresponding absolute time (e.g., for presenting global log events sequentially).
+ *
+ *
+ * @param {string[]} parseableAbsoluteStrings - a group of ISO Strings
+ * @param {boolean} reverse ascending or descending order
+ * @returns {Temporal.Absolute[]} Temporal.Absolute instance
+ */
+function getSortedInstants(parseableAbsoluteStrings, reverse = false) {
+    const sortedAbsoluteTimes = parseableAbsoluteStrings.sort(
```

Comment:
How about this?

```suggestion
    const sortedAbsoluteTimes = parseableAbsoluteStrings.map(Temporal.Absolute.from).sort(
```

Reply:
that won't work because the second argument to from() will be the index, and from will incorrectly assume this to be something else. (timezone?)
Leading to this error:
```
    return this === Absolute ? result : new this(GetSlot(result, EPOCHNANOSECONDS));
                                        ^

TypeError: this is not a constructor
```

Reply:
The second argument can be safely deleted, it does nothing and isn't supposed to be there. However this error is the result of something more subtle. The `this === Absolute` check is, I guess, supposed to check whether you are calling this method as `Temporal.Absolute.from()` or `SomeSubclass.from()`... but `parseableAbsoluteStrings.map(Temporal.Absolute.from)` overrides the `this` value that `Temporal.Absolute.from()` is called with. That's why the arrow function works, as well as any of these others:
```js
parseableAbsoluteStrings.map(v => Temporal.Absolute.from(v))
parseableAbsoluteStrings.map(Temporal.Absolute.from.bind(Temporal.Absolute))
parseableAbsoluteStrings.map(Temporal.Absolute.from, Temporal.Absolute)
```
I think this is actually a serious gotcha with the API, since `map(Temporal.Absolute.from)` seems quite natural to do.